src/rgw/rgw_auth_s3.h

```diff
@@ -114,7 +114,43 @@ class AWSAuthStrategy : public rgw::auth::Strategy,
     return aplptr_t(new decltype(apl)(std::move(apl)));
   }
 
+
 public:
+  using engine_map_t = std::map <std::string, std::reference_wrapper<const Engine>>;
+  void add_engines(const std::vector <std::string>& auth_order,
+		   engine_map_t eng_map)
+  {
+    auto ctrl_flag = Control::FALLBACK;
+    for (const auto &eng : auth_order) {
+      // The last authentication engine *must* succeed
+      if (&eng == &auth_order.back()) {
+	ctrl_flag = Control::SUFFICIENT;
```

Comment:
i think this mixes up the control flags. as i understand it, the local engine was set to FALLBACK so that if both external and local auth failed, we'd return the original error from external auth. this logic swaps the two, so that given the same order of failures, we'd return the local engine's error code instead

Reply:
Fixing, thanks